main.cpp

```diff
@@ -42,29 +109,60 @@ struct Point
 template<typename Type>
 struct Wrapper
 {
-    Wrapper(Type&& t) : val(std::move(t)) 
-    { 
-        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    Wrapper() = default;
+
+    Wrapper(Type&& t) : val(std::move(t))
+    {
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl;
     }
+    
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+    
+private:
+    Type val;
 };
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& point) : val(std::move(point))
+    {
+        
+    }
```

Comment:
Please copy the constructor body from the base template and add it to this explicit template.
- [ ] revise